system_tests/timeboost_test.go

```diff
@@ -1020,6 +1020,7 @@ func TestTimeboostBulkBlockMetadataAPIFlaky(t *testing.T) {
 
 	// A Reorg event should clear the cache, hence the data fetched now should be accurate
 	Require(t, builder.L2.ConsensusNode.TxStreamer.ReorgAt(10))
+	time.Sleep(100 * time.Millisecond) // give time for BulkBlockMetadataFetcher to receive the reorg event and clear its cache
```

Comment:
100ms seems arbitrary. Instead, can we add a loop that checks whether the cache is clear and fails after N number of retries?